src/iris/sender/coordinator.py

```diff
@@ -146,40 +103,41 @@ def update_forever(self):
                 log = logger.debug
 
             if self.is_master:
-                log('I am the sender master.')
+                log('I am the master sender')
             else:
-                log('I am not the sender master.')
+                log('I am a slave sender')
 
             metrics.set('slave_instance_count', self.slave_count)
             metrics.set('is_master_sender', int(self.is_master))
 
-            # keep track of task to purge old slave instances if i'm master; kill
-            # it otherwise
-            if self.is_master:
-                if not bool(self.prune_old_instances_task):
-                    self.prune_old_instances_task = spawn(self.prune_old_instances)
-            else:
-                if bool(self.prune_old_instances_task):
-                    self.prune_old_instances_task.kill()
-
             sleep(UPDATE_FREQUENCY)
 
-    def prune_old_instances(self):
-        while True:
-            logger.info('Purging any old instances')
-
-            connection = None
-
-            try:
-                connection = db.engine.raw_connection()
-                cursor = connection.cursor()
-                cursor.execute(REMOVE_OLD_INSTANCES_QUERY, {'old_instances_timeout': REMOVE_OLD_INSTANCES_TIMEOUT})
-                connection.commit()
-                cursor.close()
-                connection.close()
-            except Exception:
-                logger.exception('Failed purging old instances')
-                if connection:
-                    connection.close()
-
-            sleep(REMOVE_OLD_INSTANCES_FREQUENCY)
+    def leave_cluster(self):
+        self.started_shutdown = True
+        if self.party and self.party.participating:
+            logger.info('Leaving party')
+            self.party.leave()
+        if self.lock and self.lock.is_acquired:
+            logger.info('Releasing lock')
+            self.lock.release()
+
+
+class NonClusterCoordinator():
+    def __init__(self, is_master, slaves=[]):
+        self.is_master = is_master
+        if is_master:
+            logger.info('I am the master sender')
+        else:
+            logger.info('I am a slave sender')
+
+        self.slaves = cycle([(slave['host'], slave['port']) for slave in slaves])
```

Comment:
you don't need the list comp here, a generator expression will do

self.slaves = cycle((slave['host'], slave['port']) for slave in slaves)